Record move sequence

kaiweike opened the pull request:
## Description
This pull request introduces modifications to enable recording and outputting the sequence of moves.

## Changes
- Implemented `record_move` function to track each move.
- Implemented `print_moves` function to output the sequence of moves.
- Initialized `best_move` to -1 in `negamax` function. This serves as a flag to determine whether a move should be recorded.

kaiweike committed:
Record move sequence

Implemented move recording during the game and output the sequence
of moves post-game.

jserv commented on ttt.c:
```diff
@@ -3,6 +3,30 @@
 #include <stdlib.h>
 #include <string.h>
 
+#define MAX_MOVES 9 
```
Avoid hardcoded literal. Specify `(3 * 3)` instead.

kaiweike replied:
[done](https://github.com/kaiweike/ttt/commit/3384fcf9f2d71ec64ef1c91d947cfc6812434158)

jserv reviewed (changes requested):
Check https://cbea.ms/git-commit/ carefully and rework git commit messages.

kaiweike committed:
Avoid hardcoded literal

kaiweike commented:
Thanks for the suggestions. I have made the following changes:
- Updated the definition of `MAX_MOVES` to avoid hardcoded literal.
- Reworked the git commit messages.

jserv commented:
Thank @kaiweike for contributing!
